compiler/src/iree/compiler/Dialect/HAL/Conversion/StreamToHAL/Patterns.cpp

```diff
@@ -50,9 +32,37 @@ struct ContextResolveOpPattern
     auto resultTypes = llvm::to_vector(resolveOp.getResultTypes());
     assert(!resultTypes.empty() && "must have at least one result");
 
-    // TODO(multi-device): emit get with derived ordinal or lookup with attr.
-    Value device =
-        IREE::HAL::DeviceType::resolveAny(resolveOp.getLoc(), rewriter);
+    // Get the affinity from the op or an ancestor. Note that there may be no
+    // affinity specified at all.
+    auto affinityAttr = IREE::Stream::AffinityAttr::lookup(resolveOp);
+
+    // We currently only handle HAL device affinities.
+    // We could make this an interface to select the device and allow users to
+    // provide their own affinities to convert to HAL but for now we just have
+    // one way to specify device globals. Users may want to provide devices as
+    // arguments to their functions or are stored dynamically after
+    // initialization time.
```

Comment:
Grammar is clunky in the last sentence here. Reword?

Reply:
good catch